OpenRA.Mods.RA/Traits/Disguise.cs

```diff
@@ -177,6 +177,14 @@ void DisguiseAs(Actor target)
 			}
 		}
 
+		public void DisguiseAs(ActorInfo actorInfo, Player newOwner)
+		{
+			var renderSprites = actorInfo.Traits.GetOrDefault<RenderSpritesInfo>();
+			AsSprite = renderSprites == null ? null : renderSprites.GetImage(actorInfo, self.World.Map.SequenceProvider, newOwner.Country.Race);
+			AsPlayer = newOwner;
+			AsTooltipInfo = actorInfo.Traits.WithInterface<TooltipInfo>().FirstOrDefault();
+		}
```

Comment:
I guess you need to duplicate/refactor into a new method all the stuff that the other `DisguiseAs` method does in addition to changing the `As*` variables.